pontoon/translate/static/js/translate.es6

```diff
@@ -0,0 +1,499 @@
+      })
+    }).fail(() => {
+      console.log('save failed');
+    });
+  }
+
+  selectNextEntity(e) {
+    e.preventDefault();
+    if (this.state.selectedIndex < this.state.entities.length - 1) {
+      this.setState({
+        selectedIndex: this.state.selectedIndex + 1,
+      });
+    }
+  }
+
+  selectPreviousEntity(e) {
+    e.preventDefault();
+    if (this.state.selectedIndex > 0) {
+      this.setState({
+        selectedIndex: this.state.selectedIndex - 1,
+      });
+    }
+  }
+
+  toggleSidebar() {
+    this.setState((state, props) => ({
+      sidebarOpen: !state.sidebarOpen
+    }));
+  }
+}
+
+
+class LoadingScreen extends PontoonComponent {
+  render() {
+    return (
+      <div id="project-load">
+        <div className="inner">
+          <div className="animation">
+            <div></div>
+            <div></div>
+            <div></div>
+            <div></div>
+            <div></div>
+          </div>
+          <div className="text">"640K ought to be enough for anybody."</div>
+        </div>
+      </div>
+    );
+  }
+}
+
+
+/**
+ * List of entities that appears in the sidebar.
+ */
+class EntityList extends PontoonComponent {
+  render() {
+    let listBody = <h3 className="no-match"><div>ఠ_ఠ</div>No results</h3>;
+
+    if (this.props.entities.length > 0) {
+      let entities = [];
+      for (let k = 0; k < this.props.entities.length; k++) {
+        let entity = this.props.entities[k];
+        let selected = k == this.props.selectedIndex;
+
+        entities.push(
+          <EntityItem key={entity.pk}
+                      entity={entity}
+                      selected={selected}
+                      onSelectEntity={this.props.onSelectEntity} />
+        );
+      }
+
+      // Only show page mesage if we're showing an iframe.
+      let notOnPageMessage = '';
+      if (this.props.project.url) {
+        notOnPageMessage = <h2 id="not-on-page">Not on the current page</h2>;
+      }
+
+      listBody = (
+        <div className="wrapper">
+          <ul className="editables">
+            {entities}
+          </ul>
+          {notOnPageMessage}
+          <ul className="uneditables"></ul>
+        </div>
+      );
+    }
+
+    return (
+      <div id="entitylist">
+        <EntitySearch />
+        {listBody}
+      </div>
+    );
+  }
+}
+
+
+/**
+ * Search input above the list of entities in the sidebar.
+ */
+class EntitySearch extends PontoonComponent {
+  render() {
+    return (
+      <div className="search-wrapper clearfix">
+        <div className="icon fa fa-search"></div>
+        <input id="search" type="search" placeholder="Search" />
+        <div id="filter" className="select">
+          <div className="button selector all">
+            <span className="status fa"></span>
+            <span className="title">All</span>
+          </div>
+          <div className="menu">
+            <ul>
+              <li className="all"><span className="status fa"></span>All</li>
```

Comment:
I made a little [helper for Font Awesome stuff](https://github.com/mozilla/kitsune/blob/master/kitsune/community/static/community/js/contributors-common.jsx#L37-L42). It ends up looking like `<Icon name="status">`... though on second thought you might be doing something different here. I'm not familiar with this usage of Font Awesome.